src/groups/mqb/mqba/mqba_commandrouter.h

```diff
@@ -0,0 +1,177 @@
+// Copyright 2017-2023 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// mqbblp_messagegroupidmanager.h                                     -*-C++-*-
+#ifndef INCLUDED_MQBA_COMMANDROUTER
+#define INCLUDED_MQBA_COMMANDROUTER
+
+//@PURPOSE: Provide a class responsible for routing admin commands to the
+// subset of cluster nodes that should execute that command.
+//
+//@CLASSES:
+//  mqbcmd::CommandRouter: Manages routing admin commands
+//
+
+// BSL
+#include <bsl_iostream.h>
+#include <bsl_string.h>
+#include <bsl_vector.h>
+#include <bslma_managedptr.h>
+#include <bslmt_latch.h>
+#include <bslstl_sharedptr.h>
+
+// MQB
+#include <mqbcmd_messages.h>
+
+namespace BloombergLP {
+
+// FORWARD DECLARATION
+namespace bmqp_ctrlmsg {
+class ControlMessage;
+}
+namespace mqbi {
+class Cluster;
+}
+namespace mqbnet {
+class ClusterNode;
+}
+namespace mqbnet {
+template <class REQUEST, class RESPONSE, class TARGET>
+class MultiRequestManagerRequestContext;
+}
+
+namespace mqba {
+
+class RouteCommandManager {
```

Comment:
I originally called it CommandRouter, but then designed it to be a single use class (constructed per command and destroyed when the command has been processed). I did this since it has no persistent state. I tried renaming it to make that clearer (that it's disposable), though I don't think `RouteCommandManager` is much better.

Just calling it `CommandRouter` may be best.